### DISCUSSION TITLE
Support for UDMP firmware 1.12.xx

### DISCUSSION THREAD

**Discussion by kim-bjoern** [fabianishere/udm-kernel-tools#72] [Q&A] [answered]
Hi Fabian, brilliant work for the UDM "victims" around the world! Many thanks.

I've been trying to build my own UDM-kernel for the recent 1.12.2x firmware releases, and followed (to the best of my abilities) the "maintenance" instructions. My objective is to add NFS and/or FUSE modules to use with containers on my UDM Pro.

I've copied the UDM_COMMIT hash from the v1.11.4/config.mk to a 1.12.22 config, but when I compile it stops at:

snip...
dpkg-buildpackage: info: full upload; Debian-native package (full source is included)
Now running lintian --profile debian udm-kernel-tools_1.1.4_arm64.changes ...
E: udm-kernel-tools source: source-is-missing kexec-mod/kernel/kexec_mod.ko
E: udm-kernel-tools source: source-contains-unsafe-symlink kexec-mod/kernel/orig/kexec.c
E: udm-kernel-tools source: source-contains-unsafe-symlink kexec-mod/kernel/orig/kexec_core.c
E: udm-kernel-tools source: source-contains-unsafe-symlink kexec-mod/kernel/orig/kexec_internal.h
E: udm-kernel-tools source: source-is-missing kexec-mod/kernel/arch/arm64/kexec_mod_arm64.ko
E: udm-kernel-tools source: source-contains-unsafe-symlink kexec-mod/kernel/arch/arm64/orig/machine_kexec.c
E: udm-kernel-tools source: source-contains-unsafe-symlink kexec-mod/kernel/arch/arm64/orig/relocate_kernel.S
N: 1 tag overridden (1 error)
Finished running lintian.`
...snip

What am I missing? Did you (or anyone else) have any attempts to add a 1.12.xx version?

Thanks in advance.

**Comment by fabianishere** [comment, marked as answer]
You can ignore the errors and warnings from Lintian.   Did the build produce any `.deb` file?

**Reply by kim-bjoern** [reply]
Excellent! Yes, the .deb was indeed produced. 

Next, I configured and compiled a new kernel and installed it on the UDMP. It's listed with udm-bootctl, and seemed to boot. But the nfs module doesn't appear by lsmod. I'm new to kernel compilation, so I'll try again (and learn something new).

Thanks again.

**Reply by fabianishere** [reply]
You probably need to load the module first. I suppose `modprobe nfs`.

**Reply by kim-bjoern** [reply]
Thanks Fabian, I tried that

`# modprobe nfs`
`modprobe: module nfs not found in modules.dep`

I configured the NFS as y* - not as a module
I'm eager to learn how this is done, so I'll keep trying. Stay tuned.
